pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -839,35 +857,43 @@ func TestEventualConsistency(t *testing.T) {
 
 		// First we simulate NodeUpdate that should delete 'pod1'. It doesn't know about 'pod2' yet.
 		controller.NodeUpdated(item.oldNode, item.newNode)
-		// TODO(mborsz): Remove this sleep and other sleeps in this file.
-		time.Sleep(timeForControllerToProgress)
+
+		actions := fakeClientset.Actions()
+		if item.expectDelete {
```

Comment:
.Call the wait function instead of the if.

Reply:
Checking if testcase is expecting for delete action we reduce time of execution of the test, if not using it, each testcase we await until wait.Poll timeout is expired

Reply:
if you do wait.PollImmidiate it will check the condition immediately without any time interval. It could also be checked inside the utility function.